bson/datetime_ms.py

```diff
@@ -138,13 +140,17 @@ def _millis_to_datetime(millis: int, opts: CodecOptions) -> Union[datetime.datet
         seconds = (millis - diff) // 1000
         micros = diff * 1000
 
-        if opts.tz_aware:
-            dt = EPOCH_AWARE + datetime.timedelta(seconds=seconds, microseconds=micros)
-            if opts.tzinfo:
-                dt = dt.astimezone(tz)
-            return dt
-        else:
-            return EPOCH_NAIVE + datetime.timedelta(seconds=seconds, microseconds=micros)
+        try:
+            if opts.tz_aware:
+                dt = EPOCH_AWARE + datetime.timedelta(seconds=seconds, microseconds=micros)
+                if opts.tzinfo:
+                    dt = dt.astimezone(tz)
+                return dt
+            else:
+                return EPOCH_NAIVE + datetime.timedelta(seconds=seconds, microseconds=micros)
+        except ArithmeticError as err:
+            raise InvalidBSON(f"{err.args[0]} {DATETIME_ERROR_SUGGESTION}") from err
```

Comment:
What do you think about `{err}` instead of `{err.args[0]}`? I think it would result in a friendlier error message.